File: src/main/java/model/strategy/bono/NorthLookingDistanceProcessor.java
package model.strategy.bono;

import model.Coordinate;

public class NorthLookingDistanceProcessor extends DistanceProcessor
{
    public int getDistance(Coordinate actualCoordinate,
            Coordinate blockingCoordinate, Coordinate maxCoordinate)
    {
        int distance = 0;

        if (actualCoordinate.getY() < blockingCoordinate.getY()) {
            distance = getDistanceDirectly(blockingCoordinate.getY(),
                    actualCoordinate.getY());
        }
        if (actualCoordinate.getY() > blockingCoordinate.getY()) {
            distance = getDistanceClipped(blockingCoordinate.getY(),
                    actualCoordinate.getY(), maxCoordinate.getY());
        }

        return distance;
    }
}

File: snake-strategy-bono/src/main/kotlin/co/electric/snake/strategy/bonostrategy/distanceprocessor/NorthLookingDistanceProcessor.kt
package co.electric.snake.strategy.bonostrategy.distanceprocessor

import co.electric.snake.framework.model.Coordinate
import co.electric.snake.framework.model.Direction
import java.util.*

class NorthLookingDistanceProcessor : DistanceProcessor {

    override fun getDistance(direction: Direction, actualCoordinate: Coordinate, blockingCoordinate: Coordinate, maxCoordinate: Coordinate): Optional<Int> {
        var distance: Optional<Int> = Optional.empty()
        if (direction == Direction.NORTH) {
            val calculatedDistance: Int
            if (actualCoordinate.y <= blockingCoordinate.y) {
                calculatedDistance = getDistanceDirectly(blockingCoordinate.y, actualCoordinate.y)
            } else {
                calculatedDistance = getDistanceClipped(blockingCoordinate.y, actualCoordinate.y, maxCoordinate.y)
            }
            distance = Optional.of(calculatedDistance)
        }
        return distance
    }

}
